### ISSUE TITLE
Fix loadNodeConfig loses DB data

### ISSUE THREAD

**Issue by friofry** [status-im/status-go#5597]
# Problem
As mentioned in status-im/status-desktop#14929
`OverwriteNodeConfigValues` loses some data (LogLevel, Waku nodes). 
Here is an initial fix: https://github.com/status-im/status-go/pull/5527

Since `loadNodeConfig` is used in different ways in different places, this ticket should be done after https://github.com/status-im/status-go/issues/5563 (make paths in NodeConfig consistent) because `conf.DataDir = filepath.Join(b.rootDataDir, conf.DataDir)` joins 2 absolute paths in some tests.

Another issue: we load the DataDir from the db in `conf, err := nodecfg.GetNodeConfigFromDB(b.appDB)`, then
merge it with the default config `conf, err = b.OverwriteNodeConfigValues(inputNodeCfg, conf)` (it saves the merged config to the database). But only after that the `DataDir` is joined with the `conf.rootDataDir`. Some tests might fail because they contain "../static/test-account" in the DataDir (`TestLoginAndMigrationsStillWorkWithExistingDesktopUser`). 

# Implementation
Implement #5563, reuse #5527, fix tests

# Notes

from https://github.com/status-im/status-go/pull/5527#pullrequestreview-2186457625 
> But I'm pretty sure there will be problems. It affects many values and it was working before somehow.

**Comment by igor-sirotin** [comment]
I think it's time to fix this. I'm facing another issue trying to pass `TestNetworksOverride` in `CreateAccountAndLogin`:
https://github.com/status-im/status-go/blob/f660be0daa7cc742a07f6a139ea5ac9966f3ebe0/protocol/requests/create_account.go#L69

It works on the first run, but on the next Login it overwrites the saved networks with the defaults.

@friofry wanna get back to this issue and try to fix it again? 🙂
